### PR TITLE
Add upgrade test case OCP-28630

### PR DESCRIPTION
**Pull request by chao007**

### CODE REVIEW

File: features/upgrade/storage/upgrade.feature [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  Scenario: Snapshot operator should be in available status after upgrade and can created pod with snapshot
+    Given I switch to cluster admin pseudo user		
+    When I run the :create admin command with:
+      | f | <%= BushSlicer::HOME %>/testdata/storage/csi/aws-ebs-with-snapshots.yaml |
+    Then the step should succeed
+    When I use the "kube-system" project
+    And I wait up to 180 seconds for the steps to pass:
+    """
+    When I run the :describe client command with:
+      | resource | deployment         |
```
Comment (pruan-rht) [inline features/upgrade/storage/upgrade.feature:15]: @chao007 any reason behind using `oc describe` instead of `oc get`?

**Comment by pruan-rht** [comment]
please rebase

File: features/upgrade/storage/upgrade.feature (hunk 1 above)
Comment (chao007) [reply features/upgrade/storage/upgrade.feature:15]: In order to get output like this  `1 desired.*1 updated.*1 total.*1 available.*0 unavailable`

File: features/upgrade/storage/upgrade.feature [hunk 2]
```diff
@@ -139,3 +139,102 @@ Feature: Storage upgrade tests
       | ls | /mnt/storage |
     Then the output should contain:
       | test-upgrade |
+
+  # @author chaoyang@redhat.com
+  #  @upgrade-prepare
+  #@users=upuser1,upuser2
+  @admin
+  Scenario: Snapshot operator should be in available status after upgrade and can created pod with snapshot
+    Given I switch to cluster admin pseudo user
```
Comment (qinpingli) [inline features/upgrade/storage/upgrade.feature:148]: I'd like to add a check about ocp version here, only OCP version eq 4.4.0, we can run this case.

**Comment by qinpingli** [comment]
@chao007 Thanks for the automation, only one comment here, others lgtm.

File: features/upgrade/storage/upgrade.feature [hunk 3]
```diff
@@ -139,3 +139,104 @@ Feature: Storage upgrade tests
       | ls | /mnt/storage |
     Then the output should contain:
       | test-upgrade |
+
+  # @author chaoyang@redhat.com
+  #  @upgrade-prepare
+  #@users=upuser1,upuser2
```
Comment (liangxia) [inline features/upgrade/storage/upgrade.feature:145]: Bad format above.

File: testdata/storage/csi/sc.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: ebs-sc
+provisioner: ebs.csi.aws.com
+volumeBindingMode: WaitForFirstConsumer
```
Comment (liangxia) [inline testdata/storage/csi/sc.yaml:6]: I didn't see where is this file used.

File: testdata/storage/csi/aws-ebs-with-snapshots.yaml [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+      hostNetwork: true
+      containers:
+      - args:
+        - --csi-address=$(ADDRESS)
+        - --v=5
+        - --leader-election
+        env:
+        - name: ADDRESS
+          value: /var/lib/csi/sockets/pluginproxy/csi.sock
+        image: registry.svc.ci.openshift.org/ocp/4.4-art-latest:csi-external-resizer
```
Comment (liangxia) [inline testdata/storage/csi/aws-ebs-with-snapshots.yaml:342]: @qinpingli  Are we OK to expose this internal URL ?

Comment (qinpingli) [reply testdata/storage/csi/aws-ebs-with-snapshots.yaml:342]: @liangxia Good catch, thx.
@chao007 For other clusters did not have the token for `registry.svc.ci.openshift.org`, this case will be failed. Another issue is here we use the fix version `4.4`, how can we run this in other versions?

**Comment by liangxia** [comment]
Conflicting files
features/upgrade/storage/upgrade.feature

**Pull request closed without merging**